website/docs/core-concepts/writing-code/workflows.md

```diff
@@ -2,155 +2,161 @@
 
- <VideoEmbed host="youtube" videoId="iYZV9DPnugY" title="Using an API" caption="Using an API"/>  
+#### Async function data in a sync field
+When working with data in Appsmith, it's important to understand how to use async functions in sync fields. Asynchronous functions can be a powerful tool, but they can also be a source of errors if not used correctly. For more information, see [How to use data from async functions in sync fields](/learning-and-resources/how-to-guides/how-to-use-data-from-async-functions-in-sync-fields).
 
+#### Trigger actions on events
+[Actions](/reference/appsmith-framework/widget-actions) in Appsmith are predefined functions that allow you to perform specific tasks in response to user interactions or other events in your app. 
 
-To read data by using an API: Create an API - `getAllUsers` on Appsmith by adding the `API` from `Explorer` >> Click `(+)` Query/JS >> Select `New Blank API` >> Rename to `getAllUsers` (or select an existing API) Add API call to the table data property of a table widget as shown in the code snippet below:
+They can be triggered by binding them to events, such as a button click. For example, you want to execute a API on button click and show a success message on the successful execution of the API. You can bind to invoke [showAlert()](/reference/appsmith-framework/widget-actions/show-alert) function on the `onClick()` event for `onSuccess` attribute, as shown below:
 
-```
- {{getAllUsers.data}}
+```javascript
+showAlert("Data saved successfully.");
 ```
 
-:::info
-The [table data property of a table](../../reference/widgets/table/#table-data) requires a **JSON Array** so verify that your API returns a **JSON Array**.
-:::
+<VideoEmbed host="youtube" videoId="tjJIDkoCyQE" title="Global Functions" caption="Global Functions"/> 
 
-A `{{API.run()}}` method, if supplied to the [Table data](../../reference/widgets/table/#table-data), will throw an error, as table data is a Sync field and cannot perform the execution. However, you can read the response generated by the API by accessing the data property `{{API.data}}` of an API. When you bound an API data property to a widget, Appsmith executes the API **on page load**. You can modify the API settings if you wish not to execute the API **on page load**.
+You can only use the **global actions** provided out-of-the-box by Appsmith in the **async** **fields**.  
 
-**Using a JSObject**
+##### Handle success or error
 
-You have a JSObject function that fetches all Users' information, filters data for `firstname` & `email,` and then returns the result so you can bind it to the table widget and display information.
+When configuring API and query executions, it's important to provide feedback to the user about the outcome of the action. This can be done by displaying success messages when an action is completed successfully or error messages when there are issues with the execution.
```

Comment:
note that this is not just to provide feedback or messages. The onSuccess or onError callback can be used for performing any action/operation. so suggesting to remove this para.

Reply:
It's just an example. Reworded that.